### DISCUSSION TITLE
Diagnostic messages from none-ls code actions are very spammy

### DISCUSSION THREAD

**Discussion by andreykaipov** [LazyVim/LazyVim#2673] [Q&A] [answered]
Hey there! Been spending a good amount of time moving over my dotfiles over to Lua and really digging LazyVim. It's made it pretty easy to pick up. I'm currently running into a problem and my Google-fu is failing me though. Would appreciate any insight. 🙏 

After enabling the necessary Go extras:

```lua
    { import = "lazyvim.plugins.extras.lang.go" },
```

If I navigate through a Go project, I get the following messages in the bottom right corner simply by scrolling through the file:

<img width="262" alt="image" src="https://github.com/LazyVim/LazyVim/assets/9457739/b3279563-5728-40ed-8232-0113c8bba6d0">

Where does this log come from? What is it even called? I narrowed it down to the following code actions from https://www.lazyvim.org/extras/lang/go#none-lsnvim-optional:

- `nls.builtins.code_actions.gomodifytags`
- `nls.builtins.code_actions.impl`

By removing them, navigating is clean. The extra tools aren't that important, but is there a way I can still include them without the messages? I don't want to just hide the logs either because whatever the code action is doing actually stutters the UI in a jarring way.

This isn't specfic Go either, but seemingly for any codeaction from none-ls too. Does anybody else have this issue? It's driving me crazy. 😅

**Comment by jyuan0** [comment]
> I don't want to just hide the logs either because whatever the code action is doing actually stutters the UI in a jarring way.

You don't want to hide the messages although they're causing the UI to stutter? That's the way this is usually resolved (a `noice.nvim` route filter to skip the messages).

**Reply by andreykaipov** [reply]
I'm under the impression that hiding the messages would just sweep the problem under the rug and not actually prevent it from executing the code actions repeatedly.

**Reply by jyuan0** [reply]
Yes, but unfortunately that's the best "solution" that I know of. The code action is probably firing from a `CursorHold` autocmd, so silencing the messages might be the best workaround.

**Comment by andreykaipov** [comment]
So after cloning a fresh copy of the starter template, I didn't experience it. I narrowed down the problem to `nvimdev/lspsaga.nvim`, so it must be doing something odd. However without that plugin, code actions aren't recognized at all.

**Reply by jyuan0** [reply, marked as answer]
https://github.com/LazyVim/LazyVim/discussions/1178

**Reply by andreykaipov** [reply]
Oh that makes so much sense now. It's fetching all the code actions available to me so it can show me them. It's not actually executing anything. Thank you so much!! ❤️

**Reply by jyuan0** [reply]
Oh, glad you got it. I was just about to write a longer reply to clear things up.
